packages/functions/transform/src/merge-request-metrics.ts

```diff
@@ -288,65 +288,117 @@ function addUnique(newElement: number, currentArray: number[]) {
+  }).from(extract.members)
+  .where(inArray(extract.members.externalId, userIds))
+  .all();
 }
 
-async function getUserIds(timelineEvents: TimelineEventData[], extractDb: ExtractDatabase, transformDb: TransformDatabase) {
-  const reviewers: number[] = [];
-  const approvers: number[] = [];
-  const committers: number[] = [];
-  let mergedBy;
+async function getExtractUserByName(db: ExtractDatabase, name: string) {
+  return await db.select({
+    externalId: extract.members.externalId,
+    name: extract.members.name,
+    forgeType: extract.members.forgeType,
+  }).from(extract.members)
+  .where(or(
+    eq(extract.members.username, name),
+    eq(extract.members.name, name))
+  ).get()
+}
+
+async function getUserIds(timelineEvents: TimelineEventData[], extractDb: ExtractDatabase, transformDb: TransformDatabase, authorExternalId: number) {
+  const extractReviewers: number[] = [];
+  const extractApprovers: number[] = [];
+  const extractCommitters: number[] = [];
+  let mergedBy: number | undefined;
+  let extractUser;
+  let author: number | undefined = authorExternalId;
+  const usersArray: number[] = [];
+  usersArray.push(author);
+  const transformReviewers: number[] = [];
+  const transformApprovers: number[] = [];
+  const transformCommitters: number[] = [];
 
   for (const timelineEvent of timelineEvents) {
     switch (timelineEvent.type) {
       case 'reviewed':
         if (!timelineEvent.actorId) {
           break;
         }
-        const reviewer = await getId(timelineEvent.actorId, transformDb);
-        if (reviewer) {
-          addUnique(reviewer.id, reviewers);
-          if (timelineEvent.data && ((timelineEvent.data as extract.ReviewedEvent).state === 'approved')) {
-            addUnique(reviewer?.id, approvers);
-          }
+        addUnique(timelineEvent.actorId, extractReviewers);
+        addUnique(timelineEvent.actorId, usersArray);
+        if (timelineEvent.data && ((timelineEvent.data as extract.ReviewedEvent).state === 'approved')) {
+          addUnique(timelineEvent.actorId, extractApprovers);
         }
         break;
       case 'committed':
         const data = timelineEvent.data as extract.CommittedEvent;
-        const extractUserExternalId = await extractDb.select({
-          id: extract.members.externalId,
-        }).from(extract.members)
-        .where(or(
-          eq(extract.members.username, data.committerName),
-          eq(extract.members.name, data.committerName))
-        ).get();
-        if (extractUserExternalId) {
-          const committer = await getId(extractUserExternalId.id, transformDb);
-          if (committer) {
-            addUnique(committer.id, committers);
-          }
+        extractUser = await getExtractUserByName(extractDb, data.committerName) as ExtractUserArgs;
+        if (!extractUser) {
+          break;
         }
+        addUnique(extractUser.externalId, extractCommitters);
+        addUnique(extractUser.externalId, usersArray);
```

Comment:
I'm not sure if this is the right way, querying by username for each commit sounds bad. Atleast do them for unique comitters

Side-note: I wonder if we can get away with keeping repository members in memory ? In private projects we shouldn't see them that many, just a question for open-source, also big companies like google (60k engineers, citation?) probably won't use this today

https://octoverse.github.com/2022/state-of-open-source